presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoDriverUri.java

```diff
@@ -170,6 +172,12 @@ public boolean isCompressionDisabled()
         return DISABLE_COMPRESSION.getValue(properties).orElse(false);
     }
 
+    public List<Protocol> getProtocols()
```

Comment:
I suggest returning `Optional<List<Protocol>>` here (where `Optional.empty()` is returned instead of an empty list) so that someone doesn't accidentally fail to check for the empty condition in other usage sites.

Reply:
Made this `Optional` in latest push.